### DISCUSSION TITLE
How to contribute aws resource type.

### DISCUSSION THREAD

**Discussion by runluke** [cloud-custodian/cloud-custodian#7384] [General]
I want to contribute a new resource type for Athena workgroups. Do you have documentation on how to create a new AWS resource type?

**Comment by kapilt** [comment]
@tomjones2a we don't per se have docs, but we do have lots of examples in c7n/resources/ package . nutshell you define a resource class by subclassing QueryResourceManager and metadata for how to query it, what the arn is in a resource_type class attribute. then updating c7n/resources/resource_map to point to it, that later part is used as part of lazy loading mechanism to avoid excessive cold starts in lambda. we've got a few folks (~1300 ;-) hanging in chat if you want to discuss interactively. the last part is writing tests, for which we do have some content in the developer docs.
